docs/DesignDocs/DeepCopyOutParameters.md

```diff
@@ -0,0 +1,718 @@
+Support Deep Copy of Variable-Length Out Parameters
+=====
+
+This document provides a brief overview of the oeedger8r's deep copy feature and points out
+its missing piece: the support of the deep-copy `out` parameter. Next, the document describes a
+design that supports the deep-copy `out` parameter.
+
+Introduction
+------------
+
+The oeedger8r tool supports the feature of deep copy, which allows a user to pass a struct with nested pointers
+in ECALLs and OCALLs. To leverage this feature, the user should define the struct as part of an EDL file
```

Comment:
Recommend adding a reference to full_edl_serialization.md, and updating full_edl_serialization.md with a reference to this design doc.  Also full_edl_serialization.md includes the statement:

> Although EDL provides sufficient mechanisms for performing full serialization, the generator does not currently enforce these mechanisms. By default, the generator produces edge routines that implicitly copy untrusted memory references into the enclave. This results in potential security flaws and non-portable code.
> 
> To overcome these problems, the generator can be modified to warn when an EDL specification cannot be fully serialized.

I'd suggest updating that text in this PR.

Reply:
Updated. Please take a look.